### PR TITLE
feat: revamp HorizontalScroller component

### PR DESCRIPTION
**Pull request by snitin315**
Resolve https://github.com/opencollective/opencollective/issues/3783

# Description

Apply the latest design to `HorizontalScroller` component.

# Screenshots

## Desktop

https://user-images.githubusercontent.com/46647141/115136181-bc717980-a03b-11eb-95f3-4ab3a0850118.mp4

## Mobile

https://user-images.githubusercontent.com/46647141/115136187-c85d3b80-a03b-11eb-88a0-cc6bd6559f9c.mp4

### CODE REVIEW

File: components/HorizontalScroller.js [hunk 1]
```diff
@@ -1,68 +1,52 @@
...
+import { Box, Flex } from './Grid';
 
-  svg {
-    width: ${props => props.size}px;
-    height: ${props => props.size}px;
+const RefContainer = styled(Box)`
+  display: flex;
+  overflow-x: auto;
+  scroll-behavior: smooth;
+  padding: 20px;
```
Comment (Betree) [inline components/HorizontalScroller.js:16]: Adding this padding here creates some visual diffs. For example, it adds a scrollbar on https://staging.opencollective.com/babel with 1920x1080 resolution. Unless it's serving a specific purpose, can we get rid of it?

**Review by Betree — COMMENTED** [review]
Sorry, I just realized we did not link the Figma design in the issue. It should be there: https://www.figma.com/file/e71tBo0Sr8J7R5n6iMkqI42d/%5BOC.com%5D-04-Collectives?node-id=11750%3A5968. This can be helpful to check the button sizes.

This works great overall :100: The implementation is clean.

I've added some feedback below. There's also a small visual regression (double scrollbar) on `accept-financial-contributions/HostsContainer.js`:

Peek 2021-04-21 12-09

File: components/HorizontalScroller.js (hunk 1 above)
Comment (snitin315) [reply components/HorizontalScroller.js]: Yep removed 👍🏻

File: components/HorizontalScroller.js [hunk 2]
```diff
@@ -138,14 +122,18 @@ class HorizontalScroller extends React.PureComponent {
...
-        <ChevronsContainer
-          onPrevClick={canGoPrev ? this.onPrevClick : undefined}
-          onNextClick={canGoNext ? this.onNextClick : undefined}
-          {...props}
-        />
-      ),
+    return (
+      <Box>
+        <ControlsContainer mx={[2, 2, 5]} justifyContent="space-between">
+          <ArrowContainer disabled={!canGoPrev}>
```
[the remark this answers is not in the document]
Comment (snitin315) [reply components/HorizontalScroller.js:133]: Updated it to be a button with custom styles, added animation too.

**Comment by snitin315** [comment]
@Betree I have applied the latest feedback.

https://user-images.githubusercontent.com/46647141/115552876-65cd9f00-a2ca-11eb-8878-f73a02033b79.mp4

File: components/HorizontalScroller.js [hunk 3]
```diff
@@ -1,68 +1,57 @@
...
-    height: ${props => props.size}px;
+const ControlsContainer = styled(Flex)`
+  z-index: 10;
+  position: relative;
+  top: 30rem;
+`;
 
-    &:not([disabled]) {
-      color: ${themeGet('colors.primary.500')};
+const ArrowContainer = styled.button`
```
Comment (Betree) [inline components/HorizontalScroller.js:24]: Can we rather use a `StyledRoundButton`? We want to have the default button styles & docus/hover/active states for this, it's simpler to re-use existing components for that.

File: components/HorizontalScroller.js [hunk 4]
```diff
@@ -1,68 +1,57 @@
...
-      }
+  svg {
+    height: 40px;
+    padding 7px;
+    vertical-align: middle;
+    border: 1px solid #C4C7CC;
+    border-radius: 50%;
+    background-color: #FFFFFF;
+    transition: opacity 0.25s ease-in;
+    opacity: ${props => (props.disabled ? '0' : '1')};
```
Comment (Betree) [inline components/HorizontalScroller.js:37]: This is not enough, as the button stays clickable. We need to make sure it cannot be (by removing the node, or with a `display: none` for example).

Peek 2021-04-21 15-17

Comment (snitin315) [reply components/HorizontalScroller.js:37]: How about using `visibility: hidden`? it won't consume clicks.

arrow

Comment (snitin315) [reply components/HorizontalScroller.js:37]: How about manipulating `visibility: hidden` too? I have also used `StyledRoundButton` in the gif below.

arr

Comment (Betree) [reply components/HorizontalScroller.js:37]: That can work, as long as we're still able to click a link below the button it's all good.

In the preview you shared, the button colors still don't seem to match the design system. We can normally remove custom styles from there, and just use `StyledRoundButton`'s defaults.

Comment (snitin315) [reply components/HorizontalScroller.js]: Updated 👍🏻

File: components/collective-page/sections/Events.js [hunk 5]
```diff
@@ -62,50 +63,42 @@ class SectionEvents extends React.PureComponent {
...
+            </Box>
+            <ContributeCardsContainer>
+              {isAdmin && (
+                <Box px={CONTRIBUTE_CARD_PADDING_X} minHeight={150}>
+                  <CreateNew route={`/${collective.slug}/events/create`} data-cy="create-event">
+                    <FormattedMessage id="event.create.btn" defaultMessage="Create Event" />
+                  </CreateNew>
+                </Box>
+              )}
+              {upcomingEvents.map(event => (
```
Comment (SudharakaP) [inline components/collective-page/sections/Events.js:80]: I think we could simplify this duplication of identical `<Box>` elements by, 🤔 😄 

```
{[...upcomingEvents, ...pastEvents].map(event => (
    <Box key={...
```

Comment (snitin315) [reply components/collective-page/sections/Events.js]: Yeah, lets do it 🚀

**Review by Betree — APPROVED** [review]
Nice, thank you @snitin315 :muscle:

**Pull request merged**